src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -153,42 +153,41 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
       transformItems = (items: any[]) => items,
       cache = getInMemoryCache(),
     } = widgetParams || ({} as typeof widgetParams);
-    let cachedHits: InfiniteHitsCachedHits | undefined = undefined;
-    let prevState: Partial<SearchParameters>;
     let showPrevious: () => void;
     let showMore: () => void;
     let sendEvent;
     let bindEvent;
-
-    const getFirstReceivedPage = () =>
-      Math.min(...Object.keys(cachedHits || {}).map(Number));
-    const getLastReceivedPage = () =>
-      Math.max(...Object.keys(cachedHits || {}).map(Number));
+    const getFirstReceivedPage = (state: SearchParameters) => {
+      const { page = 0 } = state;
+      const pages = Object.keys(cache.read({ state }) || {}).map(Number);
```

Comment:
we probably don't want to read the cache multiple times, this function should accept the result of cache IMO

Reply:
Yeah, we could do that, but I initially didn't do it because `getShowPrevious` and `getShowMore` are calling `getFirstReceivedPage` and `getLastReceivedPage` and they don't have access to the result of cache.

Now that we kind of agree on _"do not store firstReceivedPage or lastReceivedPage in the upper scope"_ part, I've changed the implementation to provide `cachedHits` to here and there. https://github.com/algolia/instantsearch.js/pull/4534/commits/1e12727ca2a4a94a5459d6e66ba389a71f0a12e2

What do you think?

Reply:
seems all right like this for me